pkg/vulnsrc/vulnsrc.go

```diff
@@ -208,7 +211,10 @@ func (o lightOptimizer) Optimize() error {
 
 func (o lightOptimizer) lightOptimize(cveID string, tx *bolt.Tx) error {
 	// get correct severity
-	vuln := getDetailFunc(cveID)
+	vuln, isRejected := getDetailFunc(cveID)
+	if isRejected {
+		return nil
+	}
```

Comment:
In addition, we have to remove the security advisory of this CVE-ID. We have 3 types of buckets.

- vulnerability
  - includes vulnerability detail
  - it is created while optimization
  - it is used to display vulnerability information in a scan result
- vulnerability-detail
  - includes vulnerability detail, but there are duplication
  - it is removed while optimization
- security advisory (such as `Red Hat Enterprise Linux 6`)
  - includes fixed versions for vulnerability detection

We prevent rejected vulnerabilities from being inserted into the `vulnerability` bucket, but those vulnerabilities are already inserted into the `security advisory` bucket. 
https://github.com/aquasecurity/trivy-db/blob/862ec93db08b0bf995c5c0c9e6e08578f668f74b/pkg/vulnsrc/redhat/redhat.go#L144

It means they are detected without detail. We need to remove those CVE-IDs in the optimization process.

Reply:
Instead of putting and deleting the advisory, if I add one field in vulnerability detail for advisory and during optimization add advisories for CVE not rejected will it make sense?

Reply:
Can we do it even though `Advisory` is a different struct?

Reply:
Yes, we can add a field type []Advisory in vulnerability detail for a vulnerability and then we can add it in optimization

Reply:
OK, so let's go ahead with it, but I'd like to see how it's going to be. Let me review it before you'll write tests and other small fixes.

Reply:
@knqyf263 Please check, I still need to fix tests